.github/workflows/secrets-rotation-reminder.yml

```diff
@@ -0,0 +1,90 @@
+    
+          - name: Run Secrets Rotation Reminder Script
+            run: |
+              # Get the list of secret names from AWS Secrets Manager
+              secrets=$(aws secretsmanager list-secrets --region $AWS_REGION --query "SecretList[].Name" --output text)
+              
+              # Remove secrets from list that are exempt from rotation
+              delete=("environment_management" "nuke_account_ids" "nuke_account_blocklist" "mod-platform-circleci")
+              for del in ${delete[@]}
+              do
+                secrets=("${secrets[@]/$del}")
+              done
+              
+              # Define the threshold in seconds (180 days)
+              threshold=$((180 * 24 * 60 * 60))
+              
+              # Get the current timestamp in seconds
+              current_timestamp=$(date +%s)
+              
+              # Loop through each secret,  check its last changed date and raise issue if required
+              for secret in $secrets; do
+                last_changed=$(aws secretsmanager list-secret-version-ids --secret-id $secret --region $AWS_REGION --query "Versions[?contains(VersionStages,'AWSCURRENT')].CreatedDate" --output text | sort -r | head -n 1)
+                
+                # If the secret has never been changed, set the last_changed date to 0
+                if [ -z "$last_changed" ]; then
+                  last_changed=0
+                fi
+              
+                # Calculate the age of the secret in seconds
+                age=$((current_timestamp - $(date -d "$last_changed" +%s)))
+
+                # Check if there is an existing open issue to rotate the secret
+                open_issue=$(gh issue list -R ministryofjustice/modernisation-platform --search "Rotate $secret in:title" --state open)
+              
+                # Check if the secret is older than the threshold and if there is an existing open issue to rotate it, if required raise a new issue
+                if [ $age -gt $threshold ] && [ -z "$open_issue" ]; then
+                echo "$secret secret is older than 180 days (age: $((age / (24 * 60 * 60))) days)"
+                echo "Creating GitHub Issue to rotate $secret"
+                gh issue create --title "Rotate $secret Credential" --label security --body "The [secrets-rotation-reminder workflow](${{ github.server_url }}/${{ github.repository }}/actions/runs/${{ github.run_id }}) has identified that the $secret credential requires rotation as it is close to or exceeding the threshold of 180 days. 
+
+                Consult [this documentation](https://user-guide.modernisation-platform.service.justice.gov.uk/runbooks/rotating-secrets.html#how-to-rotate-secrets) which describes the process for rotation."
+                else 
+                echo "The $secret secret has not been identified for rotation (age: $((age / (24 * 60 * 60))) days)"
+                fi
+
+              done
+
+            env:
```

Comment:
this is a bit pedantic of me, but I noticed you are defining env vars globally and then in the job itself... you could argue that all of the vars are global or all of them are local to the job...there is only one job, so it probably does not matter much, but what hurts my eyes is env vars are defined in two places :) 
as I said... a bit pedantic of me :D

Reply:
That's OK - this is a symptom of me having borrowed from other jobs and examples! I will tidy that up.